react/week3/todo-app/src/components/GetTodos.js

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect } from "react";
+import TodoList from "./TodoList";
+import AddTodo from "./AddTodo"
+
+const GetTodos = () => {
+    const API_URL = "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+
+    const [todoData, setTododata] = useState([])
+    const [isLoading, setIsLoading] = useState(true);
+    useEffect(() => {
+        fetch(API_URL)
+            .then(res => res.json())
+            .then(todoArray => {
+                setIsLoading(prev => prev = false)
+                return setTododata(todoArray)
+            })
+        return () => {
+            setIsLoading((prev) => prev = true);
+        };
+    }, [])
+
+
+    //adding todo
+    const addTodo = (description, deadline) => {
+        if (description && deadline) {
+            const id = Math.floor(Math.random() * (100 - 4 + 1)) + 4;
```

Comment:
If you want to be a bit more sure that the ids are unique, I'd consider using the timestamp (maybe combined with a random number).
See this answer on stackoverflow: https://stackoverflow.com/a/8012120/7630267